web/src/search_suggestion.ts

```diff
@@ -625,10 +625,16 @@ function get_channels_filter_suggestions(last: NarrowTerm, terms: NarrowTerm[]):
     if (last.operator === "search" && common.phrase_match(last.operand, "streams")) {
         search_string = "streams:public";
     }
+    let description_html;
+    if (current_user.is_guest) {
+        description_html = "All public channels that you can view";
+    } else {
+        description_html = "All public channels";
+    }
```

Comment:
I think it might be better to spell it as `page_params.is_spectator || current_user.is_guest` consistently, like you do elsewhere. It has no functional impact, since spectators have `is_guest` set, but feels more readable to spell out the cases that we should be thinking about when choosing the string here.

Reply:
Sure, I had removed it as i found it redundant.
Done.